modules/core/src/main/scala/tethys/commons/TokenNode.scala

```diff
@@ -100,36 +100,37 @@ object TokenNode {
   }
 
   implicit class TokenNodesOps(val json: String) extends AnyVal {
-    def jsonAsTokensList(implicit producer: TokenIteratorProducer): List[TokenNode] = {
+    def jsonAsTokensList(implicit producer: TokenIteratorProducer): Either[ReaderError, List[TokenNode]] = {
       import tethys._
-      val iterator = json.toTokenIterator
-      val builder = List.newBuilder[TokenNode]
-      while(!iterator.currentToken().isEmpty) {
-        val token = iterator.currentToken()
-        val node = {
-          if (token.isArrayStart) ArrayStartNode
-          else if (token.isArrayEnd) ArrayEndNode
-          else if (token.isObjectStart) ObjectStartNode
-          else if (token.isObjectEnd) ObjectEndNode
-          else if (token.isNullValue) NullValueNode
-          else if (token.isFieldName) FieldNameNode(iterator.fieldName())
-          else if (token.isStringValue) StringValueNode(iterator.string())
-          else if (token.isNumberValue) iterator.number() match {
-            case v: java.lang.Short => ShortValueNode(v)
-            case v: java.lang.Integer => IntValueNode(v)
-            case v: java.lang.Long => LongValueNode(v)
-            case v: java.lang.Float => FloatValueNode(v)
-            case v: java.lang.Double => DoubleValueNode(v)
-            case n => NumberValueNode(n)
+      json.toTokenIterator.right.map { iterator =>
```

Comment:
This method is used only for test reason, so i think we can just throw an exception

Reply:
done